Reference/Routing/custom-controllers.md

```diff
@@ -2,167 +2,246 @@
     }
 }
 ```
+Will result in ALL requests to ANY 'product' pages being 'hijacked' and routed through your custom MVC controller!
 
-Now you can run any code that you want inside of that Action!
+The example above shows the default behaviour that Umbraco's core RenderMvcController provides - by default, the 'Index' action of the controller will be executed, and the CurrentTemplate helper sends the model containing the published content for the request to the relevant template/view. 
 
 ## Routing via template
 
-To further extend this, we've also allowed routing to different Actions based on the Template that is being rendered. By default only the Index Action exists which will execute for all requests of the corresponding document type. However, if the template being rendered is called 'HomePage' and you have an Action on your controller called 'HomePage' then it will execute instead of the Index Action. As an example, say we have a Home Document Type which has 2 allowed Templates: ‘HomePage’ and ‘MobileHomePage’ and we only want to do some custom stuff for when the ‘MobileHomePage’ Template is executed:
+A further convention is that if an action on the controller has a name that matches the template name, this action will be executed instead of the default 'Index' action.
+
+### Example hijacking route requests to a 'product' for an alternative 'AMP' template
+
+In this example, the Product Page Document Type has two templates 'ProductPage' and 'ProductAmpPage', we can hijack and handle the requests to the two templates differently.
+
+Creating the Controller as before
 
 ```csharp
-public class HomeController : Umbraco.Web.Mvc.RenderMvcController
+public class ProductPageController : Umbraco.Web.Mvc.RenderMvcController
 {
-    public ActionResult MobileHomePage(ContentModel model)
+    // Any request for the 'ProductAmpPage' template will be handled by this Action
+    public ActionResult ProductAmpPage(ContentModel model)
+    {
+        // Create AMP specific content here...
+        return CurrentTemplate(model);
+    }
+    // All other request, eg the ProductPage template will be handled by the default 'Index' action
+    public override ActionResult Index(ContentModel model)
     {
-        // Do some stuff here, the return the base Index method
-        return base.Index(model);
+        // you are in control here!
+
+        // return a 'model' to the selected template/view for this page.
+        return CurrentTemplate(model);
     }
 }
 ```
+#### How can a page be requested via two different templates?
+The page in Umbraco will have a single 'template' selected as it's default template, but it's possible to call this same page on a different template by adding ?altTemplate=othertemplatename to the Url QueryString eg:
+
+/products/superfancyproduct/?altTemplate=ProductAmpPage
+
+would request the Super Fancy Product page but force it to be loaded with the ProductAmpPage template, this can be simplified further:
```

Comment:
```suggestion
This would request the 'Super Fancy Product' page but force it to be loaded with the 'ProductAmpPage template'. This can be simplified further:
```